master/internal/resourcemanagers/fitting.go

```diff
@@ -104,11 +108,7 @@ func findDedicatedAgentFits(
 	// 2) Multi-agent tasks will receive all the slots on every agent they are scheduled on.
 	agentsByNumSlots := make(map[int][]*agentState)
 	for _, agent := range agentStates {
-		if agent.numUsedSlots() != 0 {
-			continue
-		}
-
-		constraints := []HardConstraint{labelSatisfied}
+		constraints := []HardConstraint{labelSatisfied, agentIdleSatisfied}
```

Comment:
I'm not sure about this. Doesn't this mean that if there is a zero-slot container running on an agent, it means that agent can't be used for distributed training?

Reply:
But isn't the addition of the `agentIdleSatisfied` constraint mean that if a zero-slot task is running on an agent it will not schedule a d-train task anymore?